### PR TITLE
Tidy up modeling code post-migration

### PR DESCRIPTION
**Pull request by bdewilde**
## changes

- Generally improves code quality in modeling modules, including var names, import namespaces, function signatures, and loggers (instead of print stmts)
- Refactors some pandas / matplotlib / sklearn / pytest usage w/ more modern functionality and patterns

## context

Just trying to further tidy up this new code base, and make code quality / functionality more standardized across subpackages

## questions

- Have I accidentally changed any functionality in a bad way? Besides a few tweaked function signatures, I did not intend to fundamentally change the logic.

### CODE REVIEW

**Review by skvorekn — APPROVED** [review]

File: src/student_success_tool/modeling/training.py [hunk 1]
```diff
@@ -1,89 +1,96 @@
+import logging
 import time
 
 import pandas as pd
 
+# TODO: how do we specify databricks.automl as a package dependency?
+# i.e. what's the package name??
```
[the remark this answers is not in the document]
Comment (bdewilde) [reply src/student_success_tool/modeling/training.py:7]: I _looked_ but legit could not find this out. Pretty wild in my experience.

File: src/student_success_tool/modeling/training.py [hunk 2]
```diff
@@ -1,89 +1,96 @@
...
-    if not kwargs.get("pos_label"):
-        kwargs["pos_label"] = True
-    if not kwargs.get("timeout_minutes"):
-        kwargs["timeout_minutes"] = (
-            5  # TODO: tune this! https://app.asana.com/0/0/1206779161097924/f
-        )
+    # set some sensible default arguments
+    kwargs.setdefault("pos_label", True)
+    # TODO: tune this! https://app.asana.com/0/0/1206779161097924/f
+    kwargs.setdefault("timeout_minutes", 5)
```
[the remark this answers is not in the document]
Comment (bdewilde) [reply src/student_success_tool/modeling/training.py:68]: That's correct! A perfect use case for `dict.set_default`, which I rarely get to use :)

File: src/student_success_tool/modeling/feature_selection.py [hunk 3]
```diff
@@ -151,49 +179,98 @@ def drop_collinear_features_iteratively(
...
+    LOGGER.info("dropping %s collinear features", n_features_dropped_so_far)
+
     assert all(
-        [col in features.columns for col in force_include_cols]
+        [col in df.columns for col in force_include_cols]
     ), "The dataset with selected features is missing one of the force include variables!"
-    return features
+    return df
+
+    # TODO: figure out why this below code gives different results from the original :/
```
[the remark this answers is not in the document]
Comment (bdewilde) [reply src/student_success_tool/modeling/feature_selection.py:240]: Which seed (numpy?) / what's the right call for this?

File: src/student_success_tool/modeling/feature_selection.py [hunk 4]
```diff
@@ -151,49 +179,98 @@ def drop_collinear_features_iteratively(
...
+
+    df_features = df.select_dtypes(include="number")
+    if df_features.shape[1] == 0:
+        LOGGER.warning("no numeric columns found, so no collinear features to drop")
+        return df
 
-    numeric_features = features.select_dtypes(include="number")
-    imp = SimpleImputer(missing_values=np.nan, strategy="mean")
-    numeric_features_imputed = pd.DataFrame(imp.fit_transform(numeric_features))
-    numeric_features_imputed.columns = numeric_features.columns.values
```
[the remark this answers is not in the document]
Comment (bdewilde) [reply src/student_success_tool/modeling/feature_selection.py:166]: Oops, I forgot to answer this question. It was needed because `SimpleImputer.fit_transform()` was returning a numpy array instead of a dataframe, so you needed to re-add the column names back in. I changed that in the line below this comment by leveraging `.set_output(transform="pandas")` -- now the imputer returns a dataframe with the columns already populated as you'd expect. This is a fairly new sklearn feature, and it's lovely! https://scikit-learn.org/stable/auto_examples/miscellaneous/plot_set_output.html#sphx-glr-auto-examples-miscellaneous-plot-set-output-py

File: src/student_success_tool/modeling/training.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (skvorekn) [reply src/student_success_tool/modeling/training.py:7]: Yup, same. Very confusing

File: src/student_success_tool/modeling/feature_selection.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (skvorekn) [reply src/student_success_tool/modeling/feature_selection.py:240]: Numpy is what I was thinking, although looking back at my Lee College code, I don't have a seed, so maybe I'm misremembering that that's needed
